public/javascripts/ScrabBag.js

```diff
@@ -0,0 +1,54 @@
+class ScrabBag {
```

Comment:
I love the class based approach to solving this problem.  I'm not certain I understand the individual methods that are used to solve the problem statement?  I would also consider using complete words for the name of the class - `ScrabbleBag` - to make it explicit what this object is intended to represent.

When I read the problem, I identify some critical behaviors that are necessary to solve it:
* remove( String ) - a function that takes a string of played letters (i.e. letters that have been removed from the bag)
* tiles() - a function that returns the set of tiles (it's possible that this function could be responsible for counting as well, but that doesn't really *feel like* the responsibility of a `ScrabbleBag` object).

The counting and display piece of the problem statement feels like another responsibility - I would probably create an additional function that could take the result of `.tiles()` on an instance of a `ScrabbleBag`, and partition the array of tiles into the appropriate display format.

I think these approaches would probably allow us to refactor this code a little bit to make it easier to follow, as well.